Sources/NIOOpenSSL/ByteBufferBIO.swift

```diff
@@ -0,0 +1,289 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import CNIOOpenSSL
+
+
+/// The OpenSSL entry point to write to the `ByteBufferBIO`. This thunk unwraps the user data
+/// and then passes the call on to the specific BIO reference.
+internal func openSSLBIOWriteFunc(bio: UnsafeMutablePointer<BIO>?, buf: UnsafePointer<CChar>?, len: CInt) -> CInt {
```

Comment:
- could you document that the types are silly (buf & len separate) because this function needs to be compatible with `@convention(c)`
- couldn't we accept an `UnsafeRawPointer` here?